internal/db/error_codes.go

```diff
@@ -0,0 +1,17 @@
+package db
+
+import (
+	"errors"
+
+	"github.com/jackc/pgconn"
+)
+
+// https://www.postgresql.org/docs/current/errcodes-appendix.html
+
+func IsUniqueConstraintError(err error) error {
+	var pgErr *pgconn.PgError
+	if err != nil && errors.As(err, &pgErr) && pgErr.Code == "23505" {
```

Comment:
Could `23505` be a constant?

Reply:
Oh yeah, good point. It looks much better now.